test/integration/smoke/test_templates.py

```diff
@@ -1332,7 +1332,6 @@ def test_03_deploy_vm_wrong_checksum(self):
                 domainid=self.account.domainid,
                 serviceofferingid=self.service_offering.id
             )
-            self.cleanup.append(tmpl)
```

Comment:
i'd say this, `self.cleanup.append(tmpl)` is supposed to go to line 1325,
and this line should read `self.cleanup.append(virtual_machine)`

lines 1339 and 1340 should be removed. As this code is at the moment virtual_machne will be appended to cleanup only if the create fails and `self.fail(...)` is *not* called.

a cleanup should go directly after a create (or register) and only if the remove happens for some other reason it should be removed from the list again (if a double delete would be a problem)